### PR TITLE
Reconfigure worker pod capabilities

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
1) Removing privileged and adding SYS_MODULE capability 2) Worker application in worker pod will run as User 0 in order to
   be able to run modprobe

### CODE REVIEW

File: Dockerfile.worker [hunk 1]
```diff
@@ -36,7 +36,5 @@ WORKDIR /
 
 COPY --from=builder /workspace/worker /worker
 RUN mkdir -p /mnt/img
-RUN chown 65532:65532 /mnt/img
```
Comment (qbarrand) [inline Dockerfile.worker:39]: Don't we need this?

Comment (yevgeny-shnaidman) [reply Dockerfile.worker:39]: No, since worker is not running under User 65532, then we don't need it

Comment (qbarrand) [reply Dockerfile.worker:39]: What user is it running as then?

Comment (yevgeny-shnaidman) [reply Dockerfile.worker:39]: the default one - "0"

**Comment by qbarrand** [comment]
/lgtm
Investigate further hardening in https://github.com/kubernetes-sigs/kernel-module-management/issues/538

**Pull request merged**